tests/api/api/test_submit.py

```diff
@@ -183,8 +195,11 @@ def test_submit_job_ensure_function_has_auth_set(
 
 
 def test_submit_schedule_job_from_hub_from_ui(
-    db: Session, client: TestClient, pod_create_mock, k8s_secrets_mock
+    db: Session, client: TestClient, pod_create_mock, k8s_secrets_mock, monkeypatch
 ) -> None:
+    monkeypatch.setattr(
+        mlrun.run, "extend_hub_uri_if_needed", _mock_extend_hub_uri_if_needed
```

Comment:
better mock `list_hub_sources` or `get_hub_source` if needed but the helper is part of what the test is testing